Packs/AlibabaActionTrail/Integrations/AlibabaActionTrailEventCollector/AlibabaActionTrailEventCollector.py

```diff
@@ -0,0 +1,210 @@
+        headers['x-log-date'] = headers['Date']
+        del headers['Date']
+
+        self.request.headers = headers
+
+
+class AlibabaGetEvents(IntegrationGetEvents):
+    @staticmethod
+    def get_last_run(events: list) -> dict:
+        return {'from': events[-1]['__time__']}
+
+    def _iter_events(self):
+        self.client.prepare_request()
+        response = self.call()
+        events: list = response.json()
+        events.sort(key=lambda k: k.get('__time__'))
+
+        if not events:
+            return []
+
+        while True:
+            yield events
+
+            last = events[-1]
+            self.client.set_request_filter(last['__time__'])
+            self.client.prepare_request()
+            response = self.call()
+
+            events = response.json()
+            events.sort(key=lambda k: k.get('__time__'))
+
+            if not events:
+                break
+
+
+def canonicalized_log_headers(headers):
+    content = ''
+    for key in sorted(six.iterkeys(headers)):
+        if key[:6].lower() in ('x-log-', 'x-acs-'):  # x-log- header
+            content += key + ':' + str(headers[key]) + "\n"
+    return content
+
+
+def canonicalized_resource(resource, params):
+    if params:
+        urlString = ''
+        for key, value in sorted(six.iteritems(params)):
+            urlString += u"{0}={1}&".format(
+                key, value.decode('utf8') if isinstance(value, six.binary_type) else value)
+        resource += '?' + urlString[:-1]
+        if six.PY3:
+            return resource
+        else:
+            return resource.encode('utf8')
+
+    return resource
+
+
+def base64_encodestring(s):
+    if six.PY2:
+        return base64.encodestring(s)
+    else:
+        if isinstance(s, str):
+            s = s.encode('utf8')
+        return base64.encodebytes(s).decode('utf8')
+
+
+def hmac_sha1(content, key):
+    if isinstance(content, six.text_type):  # hmac.new accept 8-bit str
+        content = content.encode('utf-8')
+    if isinstance(key, six.text_type):  # hmac.new accept 8-bit str
+        key = key.encode('utf-8')
+
+    hashed = hmac.new(key, content, hashlib.sha1).digest()
+    return base64_encodestring(hashed).rstrip()
+
+
+def get_request_authorization(resource, key, req_params, req_headers):
+    """ :return bytes (PY2) or string (PY2) """
+    content = 'GET\n\n\n'
+    content += req_headers['Date'] + "\n"
+    content += canonicalized_log_headers(req_headers)
+    content += canonicalized_resource(resource, req_params)
+    return hmac_sha1(content, key)
+
+
+def get_alibaba_timestamp_format(value):
+    timestamp: Optional[datetime]
+    if isinstance(value, int):
+        return value
+    if not isinstance(value, datetime):
+        timestamp = dateparser.parse(value)
+    if timestamp is None:
+        raise TypeError(f'after is not a valid time {value}')
+    return int(time.mktime(timestamp.timetuple()))
```

Comment:
can `value` be a `datetime` object?
If so, then the last return will raise an attribute error (NoneType has no attribute `timetuple`)
If not, then the second condition is redundant

Reply:
the value can be a string ISO format when execute `alibaba-get-events` command or `time interval` like 3 days from the integration params or Unix Timestamp (int) when get it `from getLatsRun`.  
removing the second condition